### ISSUE TITLE
Siamese/Triplet network

### ISSUE THREAD

**Issue by edgarriba** [pytorch/examples#43]
Hi! About starting to work on siamese/triplet architectures. Is there any consideration to take into account? Regarding the loss functions, is there any example using the autograd stuff?

**Comment by soumith** [comment]
if you want to implement a triplet loss, with autograd it should be fairly straightforward.

You can implement the loss by subclassing it from `nn.Module` and then defining the forward function that takes in (input, target). There's no difference in implementing a loss or a module.

These notes might help:

http://pytorch.org/docs/notes/extending.html

**Comment by apaszke** [comment]
Or you can implement it as a function, to look like the losses in the functional interface. It's also covered in the linked notes.

**Comment by edgarriba** [comment]
@soumith @apaszke I opened a PR with a first version. First I want to implement it as single example and check that all is working as expected. After that I'll integrate the loss function using the functional interface and the Phototour dataset into torchvision/datasets. Feel free to review and leave comments in https://github.com/pytorch/examples/pull/47

**Comment by apaszke** [comment]
Since you already have a PR, I'll close this issue. Let's keep the discussion in a single place.
